src/table/virtualized-table/hooks/use-dynamic-row-height.ts

```diff
@@ -103,10 +104,43 @@ const useDynamicRowHeight = ({
   );
 
   const getRowHeight = useCallback(
-    (index: number) => rowMeta.current.heights[index] ?? estimatedRowHeight,
+    (rowIdx: number) => rowMeta.current.heights[rowIdx] ?? estimatedRowHeight,
     [rowMeta, estimatedRowHeight]
   );
 
+  /**
+   * Some user actions and events can trigger row heights to be invalidated
+   * - A column is re-sized
+   * - New layout (ex. due to property changes or outside selections)
+   * - Change of page (which would load new data)
+   * - Theme change (ex. fonts)
+   * - Container element is re-sized (object re-sized in Qlik Sense or browser window re-sized)
+   */
+  useOnPropsChange(() => {
+    rowMeta.current.lastScrollToRatio = 0;
+    rowMeta.current.resetAfterRowIndex = 0;
+    rowMeta.current.heights = [];
+    rowMeta.current.totalHeight = 0;
+    rowMeta.current.count = 0;
+    rowMeta.current.measuredCells.clear();
+  }, [layout, pageInfo, getCellSize]);
+
+  const updateCellHeight = (rows: Row[]) => {
+    if (!gridState) return;
+
+    const { overscanRowStartIndex: rowStart, overscanRowStopIndex } = gridState.current;
+    const rowStop = Math.min(overscanRowStopIndex, layout.qHyperCube.qSize.qcy - 1);
+
+    for (let rowIdx = rowStart; rowIdx <= rowStop; rowIdx++) {
+      const row = rows[rowIdx] ?? {};
+      Object.values(row).forEach((cell) => {
+        if (typeof cell === 'object') {
+          setCellSize(cell.qText ?? '', rowIdx, cell.pageColIdx);
```

Comment:
To clarify a concepts here. `estimatedRowHeight` is the average height of all measured rows. There is an assumption that all other rows that we have not measured are going to be roughtly equal to the rows we already have measured. This is a crude approach and there are multiple scenarios that is just not going to be the case. But generally it still enough for `react-window` to behave nicely.

If each cell on a row gets smaller line height (3 -> 2 -> 1), the row height for that row. Will be "3 lines". So `setEstimatedRowHeight` would be called for each measured cell. But the value passed to `setEstimatedRowHeight` would not change after the "highest" cell had been measured. 

I could avoid the costly operation if I knew which column was being re-sized. Which I don't, but I think it would be possible to expose it, ex. from the `ColumnAdjuster` component.